dotnet-desktop-guide/net/wpf/events/weak-event-patterns.md

```diff
@@ -0,0 +1,128 @@
+
+## How to Implement the weak event pattern
+
+There are three ways to implement the weak event pattern. The following table lists the three approaches and provides some guidance for when to use each.
+
+| Approach | When to Implement |
+|----------|-------------------|
+| Use an existing weak event manager class | If the event you want to subscribe to has a corresponding <xref:System.Windows.WeakEventManager>, use the existing weak event manager. For a list of weak event managers that are included with WPF, see the inheritance hierarchy in the `WeakEventManager` class. Because the included weak event managers are limited, you'll probably need to choose one of the other approaches. |
+| Use a generic weak event manager class | Use a generic <xref:System.Windows.WeakEventManager%602> when an existing <xref:System.Windows.WeakEventManager> isn't available and you're looking for the easiest way to implement weak events. However, the generic `WeakEventManager<TEventSource,TEventArgs>` is less efficient than the existing or custom weak event manager because it uses reflection to discover the event from its name. Also, the code needed to register the event using the generic `WeakEventManager<TEventSource,TEventArgs>` is more verbose than using an existing or custom `WeakEventManager`. |
+| Create a custom weak event manager class | Create a custom <xref:System.Windows.WeakEventManager> when an existing `WeakEventManager` isn't available and efficiency is crucial. Although more efficient than a generic `WeakEventManager`, a custom `WeakEventManager` requires you to write more upfront code. |
+| Use a third-party weak event manager | NuGet has some [weak event managers](https://www.nuget.org/packages?q=weak+event+manager&prerel=false). Many WPF frameworks also support the pattern. For example, see [Prism's documentation on loosely coupled event subscription](https://github.com/PrismLibrary/Prism-Documentation/blob/master/docs/wpf/legacy/Communication.md#subscribing-to-events)). |
+
+The following sections describe how to implement the weak event pattern. For the generic and custom weak event manager examples, the event to subscribe to has the following characteristics.
+
+- The event name is `SomeEvent`.
+- The event is raised by the `SomeEventSource` class.
+- The event handler has type `EventHandler<SomeEventArgs>`.
+- The event passes a parameter of type `SomeEventArgs` to the event handlers.
+
+### Use an existing weak event manager class
+
+1. Find an existing weak event manager. For a list of weak event managers included with WPF, see the inheritance hierarchy of the <xref:System.Windows.WeakEventManager> class.
```

Comment:
Can we hit these up with `01.` ? so they align with the correct indentation.

```suggestion
01. Find an existing weak event manager. For a list of weak event managers included with WPF, see the inheritance hierarchy of the <xref:System.Windows.WeakEventManager> class.
```

Reply:
Done. Also removed unnecessary numbering in [this](https://review.docs.microsoft.com/en-us/dotnet/desktop/wpf/events/weak-event-patterns?view=netdesktop-6.0&branch=pr-en-us-1352#use-the-generic-weak-event-manager-class) section.